### PR TITLE
CI: Add fedora-todolist sample to E2E tests.

### PR DESCRIPTION
**Pull request by mrnold**
Run Wes' fedora-todolist VM from #1363 as a virtualization E2E test.

### CODE REVIEW

File: tests/e2e/sample-applications/virtual-machines/fedora-todolist/fedora-todolist.yaml [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+              namespace: openshift-virtualization-os-images
+            storage:
+              resources:
+                requests:
+                  storage: 30Gi
+      running: true
+      template:
+        metadata:
+          annotations:
+            vm.kubevirt.io/flavor: small
```
Comment (mateusoliveira43) [inline tests/e2e/sample-applications/virtual-machines/fedora-todolist/fedora-todolist.yaml:238]: is not better to use tiny?

**Comment by mateusoliveira43** [comment]
/test 4.14-e2e-test-kubevirt-aws
flake of restore succeeds, but application not fully working

File: tests/e2e/backup_restore_suite_test.go [hunk 2]
```diff
@@ -199,7 +199,7 @@ func runBackup(brCase BackupRestoreCase, backupName string) bool {
 	gomega.Expect(err).ToNot(gomega.HaveOccurred())
 
 	// wait for backup to not be running
-	gomega.Eventually(lib.IsBackupDone(dpaCR.Client, namespace, backupName), timeoutMultiplier*time.Minute*20, time.Second*10).Should(gomega.BeTrue())
+	gomega.Eventually(lib.IsBackupDone(dpaCR.Client, namespace, backupName), timeoutMultiplier*time.Minute*60, time.Second*10).Should(gomega.BeTrue())
```
Comment (mateusoliveira43) [inline tests/e2e/backup_restore_suite_test.go:202]: I think this is not ideal

would run in prow sometimes and see how long it takes, and put a little above

but to much above can hunt us in the future, this is also used by non VM tests

File: tests/e2e/virt_backup_restore_suite_test.go [hunk 3]
```diff
@@ -1,19 +1,32 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/openshift/oadp-operator/api/v1alpha1"
 	"github.com/openshift/oadp-operator/tests/e2e/lib"
 )
 
+var virtualMachineGVK = schema.GroupVersionResource{
```
Comment (mateusoliveira43) [inline tests/e2e/virt_backup_restore_suite_test.go:24]: for standardization, would move this to `tests/e2e/lib/virt_helpers.go` or `tests/e2e/lib/virt_storage_helpers.go` and change name to GVR

File: tests/e2e/virt_backup_restore_suite_test.go [hunk 4]
```diff
@@ -53,47 +68,103 @@ func runVmBackupAndRestore(brCase VmBackupRestoreCase, expectedErr error, update
...
+				log.Printf("Could not find VM status in API result.")
+				return false, nil
+			}
+			log.Printf("Status for VM %s/%s: %s", brCase.Namespace, brCase.Name, status)
+			return status == "Running", nil
+		})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	}
+
+	if brCase.InitDelay > 0*time.Second {
```
Comment (mateusoliveira43) [inline tests/e2e/virt_backup_restore_suite_test.go:137]: can be a TODO comment for later if we need this fast: can not we check an object status or something to see if it finished?

**Review by mateusoliveira43 — CHANGES_REQUESTED** [review]
I believe would be good to use tiny instead of small fedora image, but it seems to took "only" 20 minutes, so we can use small

File: tests/e2e/sample-applications/virtual-machines/fedora-todolist/fedora-todolist.yaml (hunk 1 above)
Comment (mrnold) [reply tests/e2e/sample-applications/virtual-machines/fedora-todolist/fedora-todolist.yaml]: I set it to tiny, we can see if it makes any difference in Prow.

File: tests/e2e/virt_backup_restore_suite_test.go [hunk 5]
```diff
@@ -158,5 +229,18 @@ var _ = ginkgov2.Describe("VM backup and restore tests", ginkgov2.Ordered, func(
 				BackupRestoreType: lib.CSIDataMover,
 			},
 		}, nil),
+
+		ginkgov2.Entry("todolist backup and restore", ginkgov2.Label("virt"), VmBackupRestoreCase{
```
[the remark this answers is not in the document]
Comment (mrnold) [reply tests/e2e/virt_backup_restore_suite_test.go]: Sure, I added more description to both tests.

File: tests/e2e/backup_restore_suite_test.go (hunk 2 above)
Comment (mrnold) [reply tests/e2e/backup_restore_suite_test.go]: I added per-test timeouts, if that's any better. When I run E2E tests locally I see backups take 30+ minutes for the Fedora VM.

File: tests/e2e/virt_backup_restore_suite_test.go (hunk 3 above)
Comment (mrnold) [reply tests/e2e/virt_backup_restore_suite_test.go]: This was already in virt_helpers, but I decided to just make GetVmStatus public instead. That removed a chunk of redundant code and got rid of this extra definition.

File: tests/e2e/virt_backup_restore_suite_test.go (hunk 4 above)
Comment (mrnold) [reply tests/e2e/virt_backup_restore_suite_test.go:106]: Sure, I put in a TODO. I haven't thought of a great way to do this yet though.

File: tests/e2e/backup_restore_suite_test.go (hunk 2 above)
Comment (mateusoliveira43) [reply tests/e2e/backup_restore_suite_test.go]: oh right, good catch

File: tests/e2e/backup_restore_suite_test.go [hunk 6]
```diff
@@ -300,7 +301,7 @@ func tearDownBackupAndRestore(brCase BackupRestoreCase, installTime time.Time, r
 
 	err = dpaCR.Delete(runTimeClientForSuiteRun)
 	gomega.Expect(err).ToNot(gomega.HaveOccurred())
-	gomega.Eventually(lib.IsNamespaceDeleted(kubernetesClientForSuiteRun, brCase.Namespace), timeoutMultiplier*time.Minute*2, time.Second*5).Should(gomega.BeTrue())
+	gomega.Eventually(lib.IsNamespaceDeleted(kubernetesClientForSuiteRun, brCase.Namespace), timeoutMultiplier*time.Minute*10, time.Second*5).Should(gomega.BeTrue())
```
Comment (mateusoliveira43) [inline tests/e2e/backup_restore_suite_test.go:304]: could be 5 as in here https://github.com/openshift/oadp-operator/pull/1381/files#diff-c3773538f578442b3a133ab0748f951d2edb007bd5e9a7d0d80091d07e98464bR126 ?

**Review by kaovilai — APPROVED** [review]
seems good to me.

File: tests/e2e/backup_restore_suite_test.go (hunk 6 above)
Comment (mrnold) [reply tests/e2e/backup_restore_suite_test.go]: Sure, done.

**Comment by mrnold** [comment]
> Since tests are fast (~30min), would be good to add CSI for CirrOS and DataMover for Fedora?

Yes, I will add these. It might have to be another pull request though, I was getting some backup failures with datamover on the Fedora test. Now that there's at least one working test, I can go back and debug it better.

**Comment by weshayutin** [comment]
@mpryc eyes when you have a moment please :)

**Review by weshayutin — APPROVED** [review]
/LGTM

**Pull request merged**